source/geod24/LocalRest.d

```diff
@@ -896,3 +975,65 @@ unittest
         writeln("Delta since sleep: ", current4 - current2);
     }
 }
+
+// Filter commands
+unittest
+{
+    __gshared C.Tid node_tid;
+
+    static interface API
+    {
+        size_t getCount();
+        void foo ();
+        void asyncFoo ();
+    }
+
+    import std.stdio;
+    static class Node : API
+    {
+        size_t count;
+        RemoteAPI!API remote;
+
+        public this()
+        {
+            this.remote = new RemoteAPI!API(node_tid);
+        }
+
+        override size_t getCount() { return this.count; }
+        override void foo () { ++this.count; }
+        override void asyncFoo()
+        {
+            runTask(
+                ()
+                {
+                    try
+                    {
+                        this.remote.foo();
+                    }
+                    catch (Exception ex)
+                    {
+                        import std.algorithm;
+                        assert(ex.msg.startsWith("Filtered command"));
```

Comment:
You could just do an `==` comparison here, no ?

Reply:
It's a bit brittle then. Any kind of tiny change will break this unittest. For example, if you add another field to `Command`. And the `Tid` which is part of the message changes between runs too.

Reply:
I meant to compare the messages only. The user code never sees `Command` anyway.